source/dlangbot/app.d

```diff
@@ -372,13 +372,17 @@ void handlePR(string action, PullRequest* _pr)
             );
 
             updateBugs([r.id], issueComment, r.fixed);
+            auto issueDesc = descs.filter!(d => d.id == r.id);
+            auto status = issueDesc.empty ? null : issueDesc.front.status;
 
-            // add event to the fixed bugzilla issues table
-            if (r.fixed)
+            // Add event to the fixed bugzilla issues table if the issue is still open.
+            // For revert and stable-merge PRs, the database will not be updated.
+            // If a PR is reverted, it is crucial that the bug is manually reopened.
+            if (r.fixed && status.among("NEW", "ASSIGNED", "REOPENED"))
             {
-                import std.algorithm.searching : find;
                 import dlangbot.database : updateBugzillaFixedIssuesTable;
-                updateBugzillaFixedIssuesTable(userName, userId, r.id, descs.find!(d => d.id == r.id)[0].severity);
+                auto severity = issueDesc.front.severity;
```

Comment:
When calculating the value of `status`, the code checks if `issueDesc` is empty, but this line does not. I believe the only case that could happen is if Bugzilla does not return information for an issue even though we requested it. I'd have to check if this is a plausible scenario. But, in either case and pedantically this could either be handled uniformly everywhere or we should just assert that `descs.length == refs.length`.

Reply:
If `issueDesc` is null then the check at line 381 will fail so this code will not be executed.

Reply:
Ah, of course.

So, this PR also fixes that. :)